File: packages/typia/src/transformers/features/protobuf/ProtobufCreateAssertDecodeTransformer.ts
import { ProtobufAssertDecodeProgrammer } from "../../../programmers/protobuf/ProtobufAssertDecodeProgrammer";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace ProtobufCreateAssertDecodeTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "protobuf.createAssertDecode",
      write: ProtobufAssertDecodeProgrammer.write,
    });
}

File: packages/transform/src/features/protobuf/ProtobufCreateAssertDecodeTransformer.go
package protobuf

import (
	coreprotobuf "github.com/samchon/typia/packages/core/src/programmers/protobuf"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var ProtobufCreateAssertDecodeTransformer = protobufCreateAssertDecodeTransformerNamespace{}

type protobufCreateAssertDecodeTransformerNamespace struct{}

func (protobufCreateAssertDecodeTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return coreprotobuf.ProtobufAssertDecodeProgrammer.Write(call.TypeText)
}
